### ISSUE TITLE
Delay starting of WebSocketClient until an attempt to connect is made

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#1516]
Evaluate why enabling the Jetty Distribution's `websocket` module starts a bunch of Jetty HttpClient threads.

**Comment by joakime** [comment]
The only reason those threads should be started, if if WebSocketClient itself has been started.
Which means that the end user application has accessed a WebSocketClient.

**Comment by joakime** [comment]
There is also a desire to have the HttpClient share the Executor with the server component.

**Comment by joakime** [comment]
The HttpClient in use by the WebSocketClient is currently following the same behaviors as the HttpClient in the proxy codebase.

**Comment by joakime** [comment]
Creating a new base and then starting Jetty with websockets does not have the HttpClient threads.

```
$ mkdir mybase
$ cd mybase
$ java -jar /path/to/jetty-home/start.jar --add-to-start=http,deploy,websocket,jmx
$ java -jar /path/to/jetty-home/start.jar
```

No HttpClient threads present.

Next up, deploying a webapp with no websocket usage ...

**Comment by joakime** [comment]
Adding a simple empty war to the new base results in HttpClient threads existing.

```
$ mkdir mybase
$ cd mybase
$ java -jar /path/to/jetty-home/start.jar --add-to-start=http,deploy,websocket
$ cp /tmp/empty.war webapps/
$ java -jar /path/to/jetty-home/start.jar
--
$ jps | grep start.jar
13172 start.jar
$ jstack -l 13172 | grep HttpClient
"HttpClient@2131952342-31" #31 prio=5 os_prio=0 tid=0x0000000029a86800 nid=0x2788 waiting on condition [0x000000002cb0e000]
"HttpClient@2131952342-30" #30 prio=5 os_prio=0 tid=0x0000000029a88000 nid=0x1b34 waiting on condition [0x000000002ca0e000]
"HttpClient@2131952342-29" #29 prio=5 os_prio=0 tid=0x0000000029a86000 nid=0x22d8 runnable [0x000000002c90e000]
"HttpClient@2131952342-28" #28 prio=5 os_prio=0 tid=0x0000000029a87800 nid=0x2fd0 waiting on condition [0x000000002c80f000]
"HttpClient@2131952342-27" #27 prio=5 os_prio=0 tid=0x0000000029b4f800 nid=0x3dc waiting on condition [0x000000002c70e000]
"HttpClient@2131952342-26" #26 prio=5 os_prio=0 tid=0x0000000029b4e800 nid=0x3b5c runnable [0x000000002c60e000]
"HttpClient@2131952342-25" #25 prio=5 os_prio=0 tid=0x0000000029b51800 nid=0x3508 runnable [0x000000002c50e000]
"HttpClient@2131952342-24" #24 prio=5 os_prio=0 tid=0x0000000029b51000 nid=0x295c runnable [0x000000002c40e000]
```

**Comment by joakime** [comment]
Fixed in commit 753ed9e
